### ISSUE TITLE
Forecasts not updated properly on cold boot and reset.

### ISSUE THREAD

**Issue by manj9501** [ThingPulse/esp8266-weather-station#178]
Whenever I cold boot my setup or reset it, only a few number of forecasts are updated, with the rest appearing blank as shown below in attached pictures

![20200229_173945](https://user-images.githubusercontent.com/36984979/75607264-3fbfcb00-5b1b-11ea-86ed-61823dd3c7e5.jpg)
![20200229_173955](https://user-images.githubusercontent.com/36984979/75607266-43ebe880-5b1b-11ea-9761-83707ae566f8.jpg)

However, after one or two updates of the forecasts after the specified UPDATE_INTERVAL_SECS, the issue gets resolved sometimes.
Can someone please point me to what I am doing wrong?

**Comment by ackbad** [comment]
Hey try installing "ESP8266 WEATHER STATION " Version 1.6.1.  Works for me now.
![15839789538793352902121146564568](https://user-images.githubusercontent.com/33710439/76480425-da49c500-63cb-11ea-9745-63c7a6841016.jpg)

**Comment by manj9501** [comment]
I was actually using version 1.6.6 of the library in December. 
Back then, things were absolutely fine.

**Comment by marcelstoer** [comment]
You are invited to test the fix that @reiyawea provided in #179.

**Comment by manj9501** [comment]
Just tested the fix provided by @reiyawea. And I'd say the probability of timeout happens to be more than 90% on my color display with nodeMCU setup. See the logs attached.

```
Connecting to WiFi Terrace/9620666222
..........Connected...Initializing touch screen...
Mounting file system...
SPIFFS opened: OK
.
Current time: 1584315588
Getting url: http://api.openweathermap.org/data/2.5/weather?id=1277333&appid=4c27f8d598e23e3fb84398e90a049bb1&units=metric&lang=en
[HTTP] GET...
[HTTP] GET... code: 200
start document
Getting url: http://api.openweathermap.org/data/2.5/forecast?id=1277333&appid=4c27f8d598e23e3fb84398e90a049bb1&units=metric&lang=en
[HTTP] GET...
[HTTP] GET... code: 200
start document
lost in client with a timeout

 ets Jan  8 2013,rst cause:2, boot mode:(3,7)

load 0x4010f000, len 1392, room 16 
tail 0
chksum 0xd0
csum 0xd0
v3d128e5c
~ld
```

As a side note, am yet to test this fix on my I2C display setup. Will update in a few hours from now.

**Comment by manj9501** [comment]
Exactly similar behavior observed on I2C display setup with nodeMCU.
Resets after the **Updating forecasts...** screen (shown below) and the same process repeats.
![20200316_000057](https://user-images.githubusercontent.com/36984979/76708044-e90ce080-6719-11ea-8535-84c50a1fbfd4.jpg)

The debugging log is completely identical to the one I posted for the color display setup.

**Comment by Peter-Zhifei** [comment]
I have tested version 1.6.6, 1.6.5 and 1.6.4 library,the monitor works and http calls back 200.
But I used version 2.0 ,http calls error code likes -11 , -00 and so on.
So I think the V2.0 library has bugs.

**Comment by Peter-Zhifei** [comment]
1.6.6 version can't get the proper data now.

**Comment by reiyawea** [comment]
It seems that the HTTP library has some trouble dealing with HTTP response with large payload. Maybe we shall try using TCP socket directly, which may be tough to code, but hopefully memory-efficient.

**Comment by marcelstoer** [comment]
It's not tough to code at all - at least not for basic HTTP operations. True, it's a major PITA for things like multi-part file upload over HTTPs as [we experienced first-hand](https://github.com/ThingPulse/iotworkshop2019/blob/fileUpload/src/main.cpp#L165). I argue that the code in the fix I will propose is at least as legible as the current version. Let's see how that turns out once I'm done. Give me a few days.

**Comment by marcelstoer** [comment]
Based on the discussions here and in #179 I created a new PR #180 that supersedes #179. I encourage everyone to test that one so we confirm it also works outside my test bed.  The essentials of the new code: https://github.com/ThingPulse/esp8266-weather-station/blob/9a59df23bca8e35fcd6f7a4ee0436b4565fcc33a/src/OpenWeatherMapForecast.cpp#L55-L87

**Comment by manj9501** [comment]
I've tested the code with both my SPI color LCD as well the I2c OLED display and I can say that the changes made in the PR #180  have caused both the setups to work just fine. 

Also, during my testing, I observed no timeouts as such, so as @reiyawea  believes, it must have something to do with bad network condition.

**Comment by marcelstoer** [comment]
@manj9501 Thanks for testing so thoroughly and the encouraging feedback.

**Comment by marcelstoer** [comment]
The fix was shipped with v2.0.1: https://twitter.com/thingpulse/status/1242136395336036360

**Comment by manj9501** [comment]
I've updated to the latest and it works like a charm!
Thanks for the prompt intervention!
